stdlib/src/builtin/bool.mojo

```diff
@@ -232,6 +233,17 @@ struct Bool(
         """
         return __mlir_op.`pop.select`[_type=Int](self.value, Int(1), Int(0))
 
+    @always_inline("nodebug")
+    fn __float__(self) -> Float64:
+        """Convert this Bool to a float.
+
+        Returns:
+            1.0 if True else 0.0 otherwise.
+        """
+        return __mlir_op.`pop.select`[_type=Float64](
+            self.value, Float64(1.0), Float64(0.0)
+        )
```

Comment:
Could it be that this causes the recursive reference somehow?

Reply:
I don't think so because `Int` was the first thing I tried to implement it on, so the only other changes were adding the trait itself at the time.

Reply:
We can fix this up after it lands, but we can use the new `_select_register_value` I added at https://github.com/modularml/mojo/blob/965c59196977aedac475cc6367071fbb3de6c824/stdlib/src/utils/_select.mojo